### ISSUE TITLE
Failed to post comment

### ISSUE THREAD

**Issue by nicolasbock** [canonical/athena-core#177]
After some time the Salesforce session ID expires and Athena fails to post comments to cases. The trace of this behavior in the log is

```
2024-07-14 18:11:11 [info]: Processing comment for case 500N100000EiSQLIA3
2024/07/14 18:11:11 [simpleforce] request failed, 401
2024/07/14 18:11:11 [simpleforce] Failed resp.body:  [{"message":"Session expired or invalid","errorCode":"INVALID_SESSION_ID"}]
2024/07/14 18:11:11 [simpleforce] failed to process http request, [simpleforce] Error. http code: 401 Error Message:  Session expired or invalid Error Code: INVALID_SESSION_ID
2024-07-14 18:11:11 [error]: Failed to post comment to case id: 500N100000EiSQLIA3
2024-07-14 18:11:11 [info]: Successfully posted comment on case 500N100000EiSQLIA3 for 1 reports
```

Athena should notice such failures and try a little harder to recover. At the very least, the comment that was going to be posted should be stored so that it can be posted once the connection is restored.

**Comment by mustafakemalgilor** [comment]
The last log line is interesting:

```
Successfully posted comment on case 500N100000EiSQLIA3 for 1 reports
```

It seems like athena somehow thinks it successfully posted a comment where it didn't.

**Comment by nicolasbock** [comment]
This is hopefully fixed now with #198 and #189
